src/tasks/DockerTaskProvider.ts

```diff
@@ -40,6 +41,9 @@ export abstract class DockerTaskProvider implements TaskProvider {
                 actionContext.errorHandling.suppressDisplay = true; // Suppress display. VSCode already has a modal popup and we don't want focus taken away from Terminal window.
                 actionContext.errorHandling.rethrow = true; // Rethrow to hit the try/catch outside this block.
 
+                // eslint-disable-next-line no-unused-expressions, @typescript-eslint/no-floating-promises
+                ext.ams?.recordActivity('overallnoedit');

```

Comment:
Can the ams internally handle the user opt-in. Otherwise the caller needs to remember it may not be initialized and user ?.

Reply:
Changed the implementation to handle opt-in internally for both AMS and experimentation service.